source/adapters/hip/image.cpp

```diff
@@ -8,178 +8,1335 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <hip/hip_runtime.h>
+
+#include "common.hpp"
+#include "context.hpp"
+#include "enqueue.hpp"
+#include "event.hpp"
+#include "image.hpp"
+#include "logger/ur_logger.hpp"
+#include "queue.hpp"
+#include "sampler.hpp"
 #include "ur/ur.hpp"
+#include "ur_api.h"
+
+ur_result_t urCalculateNumChannels(ur_image_channel_order_t order,
+                                   unsigned int *NumChannels) {
+  switch (order) {
+  case ur_image_channel_order_t::UR_IMAGE_CHANNEL_ORDER_A:
+  case ur_image_channel_order_t::UR_IMAGE_CHANNEL_ORDER_R:
+    *NumChannels = 1;
+    return UR_RESULT_SUCCESS;
+  case ur_image_channel_order_t::UR_IMAGE_CHANNEL_ORDER_RG:
+  case ur_image_channel_order_t::UR_IMAGE_CHANNEL_ORDER_RA:
+    *NumChannels = 2;
+    return UR_RESULT_SUCCESS;
+  case ur_image_channel_order_t::UR_IMAGE_CHANNEL_ORDER_RGB:
```

Comment:
Is there a reason that `RGB` is kept separate from the other unsupported formats?

Reply:
RGB is a 3-channel format which is not supported by either Cuda or HIP. I think currently only Level Zero supports this. Hence, it's handled as a separate case here.